### DISCUSSION TITLE
About using imodel.js front-end to create and edit imodel

### DISCUSSION THREAD

**Discussion by shenshengyi** [iTwin/itwinjs-core#121] [General] [answered]
hi, I want to learn the current method of using imodel.js front-end to create and edit imodels and synchronize them to the cloud. Because there is a demand now, new model elements need to be added to the existing imodel. Is there an effective way to achieve it using imodel.js-front-end?

**Comment by pmconne** [comment, marked as answer]
Hi shenshengyi,

Editing iModels involves pulling a briefcase (local copy of the iModel), making changes to it, and then pushing those changes back to the cloud - not too different than working with a distributed source control system like github. We have some unstable APIs for doing this kind of thing, but they are under active development as we put together a comprehensive framework for authoring applications. In the meanwhile, if you can be more specific about what you want to accomplish we can point you in the right direction and take into account your use cases as we develop the framework.

Thanks,
Paul

**Reply by shenshengyi** [reply]
Hi,paul.

I need to carry on some simple geometry on existing imodel create edit operation, now I have a preliminary implementation, can also be right to save and push to the cloud, but the process may not be perfect, for example, I create geometry elements such as on their own can be displayed in the view state, rather than on a noticeable imodel existing in view state.

My actual scenario is as follows: The railway model needs to check whether the location of the signal lamp is reasonable. Since no point light source is found in imodel.js, simple geometry is used to simulate the light to check whether the light will form shadows with the obstacle, so as to judge whether the location of the signal lamp is correct.
